platforms/jvm/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/detection/DefaultTestScanner.java

```diff
@@ -67,8 +75,24 @@ private void filenameScan() {
         candidateClassFiles.visit(new ClassFileVisitor() {
             @Override
             public void visitClassFile(FileVisitDetails fileDetails) {
-                TestClassRunInfo testClass = new DefaultTestClassRunInfo(getClassName(fileDetails));
-                testClassProcessor.processTestClass(testClass);
+                TestDefinition testDefinition = new ClassTestDefinition(getClassName(fileDetails));
+                testClassProcessor.processTestDefinition(testDefinition);
+            }
+        });
+        candidateDefinitionDirs.forEach(dir -> {
+            if (dir.exists()) {
+                if (dir.isDirectory()) {
+                    if (dir.canRead()) {
```

Comment:
Looking at the tests, this cannot ever be hit anyways because it would imply that we have non-readable inputs.  The task would never be started